### DISCUSSION TITLE
Firebase cloud functions alternative

### DISCUSSION THREAD

**Discussion by tstreamDOTh** [supabase/supabase#691] [Questions] [answered]
How do you plan to implement the firebase cloud function alternative? Do you plan to consider something like https://github.com/openfaas/faas

**Comment by kiwicopple** [comment, marked as answer]
Hey @tstreamDOTh - we haven't released a full "functions" implementation. For now the best way to do this is to do one of the following:

- create stored procedures and call them with [`rpc()`](https://supabase.io/docs/client/rpc), or use Postgres triggers
- put your functions into your app (eg, using [Next.js](https://nextjs.org/docs/api-routes/introduction) or using [Netlify](https://functions.netlify.com/))
- create them separately, using something like Serverless, or OpenFaaS

We will release Functions in Q2 this year 👍

**Reply by panmona** [reply]
Am I right in assuming that your functions implementation will be hosted on AWS aswell?
Do you know how difficult it will be to move already made functions with for exmaple Lambda, Azure or Google Cloud to the Supabase function implementation?

**Reply by inian** [reply]
Hey @panmau, at the start, we will let you bring in your own AWS Lambda functions. Over time we will offer support for functions in other cloud environments and also provide a functions implementation managed by Supabase (built on something like OpenFaaS)

**Reply by panmona** [reply]
Thanks for the quick answer. That is very helpful to know :+1:
